fmf/plugins/pytest/plugin.py

```diff
@@ -0,0 +1,310 @@
+    merge_minus_list = merge_minus_list or config.get(CONFIG_MERGE_MINUS, [])
+    doc_str = (test.method.__doc__ or "").strip("\n")
+    # set summary attribute if not given by decorator
+    current_name = __get_fmf_attr_name(test.method, SUMMARY_KEY)
+    if not hasattr(test.method, current_name):
+        # try to use first line of docstring if given
+        if doc_str:
+            summary = doc_str.split("\n")[0].strip()
+        else:
+            summary = (
+                (os.path.basename(filename) + " " if filename else "")
+                + (cls.name + " " if cls.name else "")
+                + test.name
+                )
+        setattr(test.method, current_name, summary)
+
+    # set description attribute by docstring if not given by decorator
+    current_name = __get_fmf_attr_name(test.method, DESCRIPTION_KEY)
+    if not hasattr(test.method, current_name):
+        # try to use first line of docstring if given
+        if doc_str:
+            description = doc_str
+            setattr(test.method, current_name, description)
+    # generic FMF attributes set by decorators
+    for key in TMT_ATTRIBUTES:
+        # Allow to override key storing with merging postfixes
+        override_postfix = ""
+        if key in merge_plus_list:
+            override_postfix = "+"
+        elif key in merge_minus_list:
+            override_postfix = "-"
+        __update_dict_key(
+            test.method,
+            fmf_prefixed_name(key),
+            key,
+            test_dict,
+            override_postfix,
+            )
+
+    # special config items
+    if CONFIG_ADDITIONAL_KEY in config:
+        for key, fmf_key in config[CONFIG_ADDITIONAL_KEY].items():
+            __update_dict_key(test.method, key, fmf_key, test_dict)
+    if CONFIG_POSTPROCESSING_TEST in config:
+        __post_processing(
+            test_dict, config[CONFIG_POSTPROCESSING_TEST], cls, test, filename
+            )
+    return test_dict
+
+
+def define_undefined(input_dict, keys, config, relative_test_path, cls, test):
+    for item in keys:
+        item_id = "/" + item
+        default_key(input_dict, item_id, empty_obj={})
+        input_dict = input_dict[item_id]
+    test_data_dict(
+        test_dict=input_dict,
+        config=config,
+        filename=relative_test_path,
+        cls=cls,
+        test=test,
+        )
+
+
+def collect(opts):
+    plugin_col = ItemsCollector()
+    pytest.main(
+        ["--collect-only", "-pno:terminal", "-m", ""] + opts,
+        plugins=[plugin_col])
+    for item in plugin_col.items:
+        func = item.function
+        for marker in item.iter_markers():
+            key = marker.name
+            args = marker.args
+            kwargs = marker.kwargs
+
+            if key == "skip":
+                TMT.enabled(False)(func)
+            elif key == "skipif":
+                # add skipif as tag as well (possible to use adjust, but
```

Comment:
Could skipif be evaluated at this point? Imho `mark.skipif(False)` should end up as `enabled: False` and not `tag: SKIP False`

Reply:
It is not possible, because evaluation will be done in sense of TMT on `guest`, although TMT discovery is done on `host` system.
This `skipIf` is little bit problem to incorporate somehow easily into FMF metadata, but this tag was the simplest solution and probably the easy to understand.
Probably the beast one could be to trasfer it to something like:
```
adjust:
    when:
      interpreter: python
      code: False
    enabled: False
```
and execute adjust condition on guest system. But nothing similar is supported right now.